tls-tutorial/doc/use_your_own_certificate.md

```diff
@@ -0,0 +1,685 @@
+   i:CN = TLS Tutorial CA, C = DE, L = Bavaria, O = Tutorial Organization
+```
+
+As you can see the chain has two links. The server certificate (denoted with index zero) and the issuing CA (index one).
+
+The abbreviation `s:` stands for "subject name", `i:` for "issuer". You can also see that the CA certificate is self-signed.
+
+We also learn a few details about the cipher and server public key. This helps us judge, if the cryptography used is still secure enough:
+
+```
+New, TLSv1.3, Cipher is TLS_AES_256_GCM_SHA384
+Server public key is 4096 bit
+```
+
+At the time of this writing TLS 1.3 (January 2023), AES 256 and RSA 4096 are all considered secure.
+
+As mentioned before, asymmetric cryptography is only used to establish the initial connection. Once that is done the TLS communication partners negotiate a symmetric session. You can see this here: 
+
+```
+Post-Handshake New Session Ticket arrived:
+SSL-Session:
+    Protocol  : TLSv1.3
+    Cipher    : TLS_AES_256_GCM_SHA384
+    Session-ID: DAB72937942090F3588FD90368047E7B90361EAE4427DD326530F7CE6041174E
+    Session-ID-ctx: 
+    Resumption PSK: F2055F6F181E774D3B05EBF321662C702F32DCB48D057B9C928FF35FC217C5A167773EF4715339E34B5061C113511A00
+    PSK identity: None
+    PSK identity hint: None
+    SRP username: None
+    TLS session ticket lifetime hint: 300 (seconds)
+    TLS session ticket:
+    0000 - 4b 25 3d [...]
+```
+
+[Session Tickets](#rfc5077) are a way to allow resuming sessions without keeping server-side state per client. If you imagine how many clients a TLS-capable server might have to handle under high load, this is quite useful.
+
+The most important part here are the cipher and the ticket lifetime of 5 minutes.
+
+**Side-note:** if you ever experience TLS connections breaking of after a certain time, you should check first this session ticket lifetime and then any intermediate network components like firewalls. Sometimes they interfere with the connection renewal process.
+
+Finally, you can connect the MySQL client to the server using TLS:
+
+```shell
+mysql -u tutorial_user -p --ssl-mode=REQUIRED -u tutorial_user
+```
+
+The source side of our import via TLS is now set up and ready to use.
+
+## Setting up Docker
+
+For the next steps we need a Docker setup
+
+1. Install Docker
+    ```shell
+    sudo apt install docker.io
+    ```
+2. Add your current user to the `docker` group
+    ```shell
+    sudo usermod -aG docker "$USER"
+    ```
+3. Log out and back in to apply the group change
+4. Verify that your current user is in the `docker` group
+    ```shell
+   groups
+    ```
+
+## Installing Exasol via Docker
+
+Exasol's [docker-db](https://github.com/exasol/docker-db) provides a Docker container that is quite convenient for experiments like our tutorial.
+
+The following docker command downloads the container's layers and afterwards starts the container. Expect a couple of minutes until the download and database start are done.
+
```

Comment:
First you need to start the docker daemon with something like sudo dockerd 
We need to explain how the Exasol docker container can be safely shutdown and restarted in case people don’t do the whole tutorial in one go.